ports/mimxrt10xx/boards/feather_m7_1011/board.c

```diff
@@ -26,21 +26,40 @@
  */
 
 #include "supervisor/board.h"
-#include "boards/flash_config.h"
-#include "mpconfigboard.h"
 #include "shared-bindings/microcontroller/Pin.h"
 
-void board_init(void) {
-    // SWD Pins
-    common_hal_never_reset_pin(&pin_GPIO_AD_13);// SWDIO
-    common_hal_never_reset_pin(&pin_GPIO_AD_12);// SWCLK
+// These pins should never ever be reset; doing so could interfere with basic operation.
+STATIC const mcu_pin_obj_t *_reset_forbidden_pins[] = {
+    &pin_GPIO_AD_13,// SWDIO
+    &pin_GPIO_AD_12,// SWCLK
 
     // FLEX flash
-    common_hal_never_reset_pin(&pin_GPIO_SD_12);
-    common_hal_never_reset_pin(&pin_GPIO_SD_11);
-    common_hal_never_reset_pin(&pin_GPIO_SD_10);
-    common_hal_never_reset_pin(&pin_GPIO_SD_09);
-    common_hal_never_reset_pin(&pin_GPIO_SD_08);
-    common_hal_never_reset_pin(&pin_GPIO_SD_07);
-    common_hal_never_reset_pin(&pin_GPIO_SD_06);
+    &pin_GPIO_SD_12,
+    &pin_GPIO_SD_11,
+    &pin_GPIO_SD_10,
+    &pin_GPIO_SD_09,
+    &pin_GPIO_SD_08,
+    &pin_GPIO_SD_07,
+    &pin_GPIO_SD_06,
+};
+
+STATIC bool _reset_forbidden(const mcu_pin_obj_t *pin) {
```

Comment:
How about factoring this code out instead of copy and pasting it? Should be easy if you pass the pin array in.

Reply:
good idea to moving this into Pin.c or something and using a weak empty pin array.